packages/router/src/adapters/contract/contract.ts

```diff
@@ -676,127 +666,24 @@ export const cancelRouterContract = async (
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        chainId === txData.sendingChainId ? TransactionReasons.CancelSender : TransactionReasons.PrepareReceiver,
-        requestContext,
-      );
-
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      logger.warn("Router contract cancel: Gelato send failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  // 2. If gelato is not supported, or gelato send failed, try using the router network.
-  if (useRelayer) {
-    logger.info("Router contract cancel: sending using router network", requestContext, methodContext, {
-      cancelParams,
-      routerRelayerFeeAsset,
-      routerRelayerFee,
-    });
-
-    try {
-      const payload = {
-        chainId,
-        to: routerContractAddress,
-        type: MetaTxTypes.RouterContractCancel,
-        data: {
-          params: cancelParams,
-          signature,
-          relayerFee: routerRelayerFee,
-          relayerFeeAsset: routerRelayerFeeAsset,
-        } as MetaTxPayloads[typeof MetaTxTypes.RouterContractCancel],
-      };
-      await messaging.publishMetaTxRequest(payload);
-
-      // listen for event on contract
-      const { event } = await cancelEvt
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        chainId === txData.sendingChainId ? TransactionReasons.CancelSender : TransactionReasons.PrepareReceiver,
-        requestContext,
-      );
-
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      // NOTE: It is possible that the actual error was in the subscriber, and the above event's timeout
-      // (see waitFor) is the error we actually caught in this block.
-      logger.warn("Router contract cancel: router network failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  logger.info("Router contract cancel: sending using txservice", requestContext, methodContext, { cancelParams });
-  const receipt = await txService.sendTx(
-    onchainTx,
-    requestContext,
-  );
-
-  incrementGasConsumed(
-    chainId,
-    receipt,
-    txData.sendingChainId === chainId ? TransactionReasons.CancelSender : TransactionReasons.CancelReceiver,
+      relayerFee: useRelayer
+        ? {
+            asset: routerRelayerFeeAsset,
+            amount: routerRelayerFee,
+          }
+        : undefined,
+      reason: chainId === txData.sendingChainId ? TransactionReasons.CancelSender : TransactionReasons.CancelReceiver,
```

Comment:
Could still be a `metaTx` relay, as seen [here](https://github.com/connext/nxtp/blob/cdaccfdc3018803e2df6bba26e6b77f969de6582/packages/router/src/lib/operations/metaTx.ts#L251)